### ISSUE TITLE
Sorter in `boutique-inventory` for equal prices

### ISSUE THREAD

**Issue by jiegillet** [exercism/elixir#928]
I was going through `boutique-inventory`, defined the first function as
```elixir
  def sort_by_price(inventory) do
    Enum.sort(inventory, fn item1, item2 -> item1.price < item2.price end)
  end
```
And a test failed with

```elixir
 1) test sort_by_price/1 sorts items by price (BoutiqueInventoryTest)
     test/boutique_inventory_test.exs:10
     Assertion with == failed
     code:  assert BoutiqueInventory.sort_by_price([%{price: 65, name: "Maxi Yellow Summer Dress", quantity_by_size: %{}}, %{price: 60, name: "Cream Linen Pants", quantity_by_size: %{}}, %{price: 33, name: "Straw Hat", quantity_by_size: %{}}, %{price: 60, name: "Brown Linen Pants", quantity_by_size: %{}}]) == [%{price: 33, name: "Straw Hat", quantity_by_size: %{}}, %{price: 60, name: "Cream Linen Pants", quantity_by_size: %{}}, %{price: 60, name: "Brown Linen Pants", quantity_by_size: %{}}, %{price: 65, name: "Maxi Yellow Summer Dress", quantity_by_size: %{}}]
     left:  [
              %{name: "Straw Hat", price: 33, quantity_by_size: %{}},
              %{name: "Brown Linen Pants", price: 60, quantity_by_size: %{}},  # <== This one is flipped with
              %{
                name: "Cream Linen Pants",                                     # <== this one
                price: 60,
                quantity_by_size: %{}
              },
              %{
                name: "Maxi Yellow Summer Dress",
                price: 65,
                quantity_by_size: %{}
              }
            ]
     right: [
              %{name: "Straw Hat", price: 33, quantity_by_size: %{}},
              %{
                name: "Cream Linen Pants",
                price: 60,
                quantity_by_size: %{}
              },
              %{name: "Brown Linen Pants", price: 60, quantity_by_size: %{}},
              %{
                name: "Maxi Yellow Summer Dress",
                price: 65,
                quantity_by_size: %{}
              }
            ]
     stacktrace:
       test/boutique_inventory_test.exs:12: (test)
```

My solution is sorted by price, but behavior for same price is not defined. Changing `<` to `<=` in `Enum.sort` fixed the test, but the problem remains. I can see two ways of solving it

1. Change prices in the test so no two are the same
2. Specify a secondary sorting (alphabetical order of `name`)?

I'm leaning towards 1. Thoughts?

**Comment by angelikatyborska** [comment]
Neither? :) I don't think this is a bug. Usually when you try to sort by values that are equal, you want the sorting to be stable, meaning not change the original order. This is exactly what this test checks. Maybe it could be an improvement to make this explicit? E.g. change the current test to say `test "the order of equal values doesn't change"` and add before that a simpler checks with all different values?

The hints contain a reference to the correct function to use in this task, and it's not `Enum.sort` :D

**Comment by jiegillet** [comment]
>The hints contain a reference to the correct function to use in this task, and it's not Enum.sort :D

Hints are for losers!! 🙈 
Seriously though, I have no excuse, I have used `sort_by` before... Actually I first used `Enum.sort(inventory, fn %{price: price} -> price end)` thinking that it would be a cool design decision if using a function of arity one would behave like `sort_by`. Unfortunately not.

 Changing the test title is good idea. I think I'll do that.
